src/proxy/socks5.rs

```diff
@@ -0,0 +1,176 @@
+use crate::config::Config;
+use crate::proxy::outbound::OutboundConnection;
+use crate::proxy::Error;
+use anyhow::Result;
+use net::Ipv4Addr;
+use std::net;
+use byteorder::{BigEndian, ByteOrder};
+use crate::workload::WorkloadInformation;
+use crate::{identity, socket};
+use std::net::{IpAddr, SocketAddr};
+use tokio::io::AsyncReadExt;
+use tokio::net::{TcpListener, TcpStream};
+use tracing::{error, info, warn};
+use tokio::io::AsyncWriteExt;
+
+pub struct Socks5 {
+    cfg: Config,
+    cert_manager: identity::SecretManager,
+    workloads: WorkloadInformation,
+    listener: TcpListener,
+}
+
+impl Socks5 {
+    pub async fn new(
+        cfg: Config,
+        cert_manager: identity::SecretManager,
+        workloads: WorkloadInformation,
+    ) -> Result<Socks5, Error> {
+        let listener: TcpListener = TcpListener::bind("127.0.0.1:15080")
+            .await
+            .map_err(Error::Bind)?;
+        match socket::set_transparent(&listener) {
```

Comment:
don't think we need this, its for TPROXY

Reply:
The joy of cut&pasting... 

BTW - the code seems to use 'orig_dst_addr' - AFAIK this is not working for TPROXY, only for REDIRECT ( with TPROXY - it's the local address ). I'll try to add the shell script that sets TPROXY with a single namespace and test.